whistleblower/whistleblower_bot/process_event.py

```diff
@@ -0,0 +1,191 @@
+    else:
+        if "purge config" in message:
+            _purge_user_config(event)
+        elif "get config" in message:
+            _get_user_config(event)
+        else:
+            _chat_post_message(event.get("channel"), utils.HELP_MESSAGE)
+
+
+def _get_user_config(event):
+    user = event.get("user")
+    if not user:
+        logger.warning("User is not present in incomming event, could not retrieve config")
+        return
+    if utils.redis_client.exists(user):
+        user_conf = json.loads(utils.redis_client.get(user))
+        _ack_config_get_msg(user_conf["phrases"], user_conf["channels"], event.get("channel"))
+    else:
+        _chat_post_message(event.get("channel"), message="You don't have any configuration!")
+
+
+def _ack_config_get_msg(phrases, channels, chan_post, base_msg=None):
+    msg = utils.DISPLAY_CONF_MESSAGE.format(
+        "\n".join(phrase for phrase in phrases), " ".join(channels)
+    )
+    if base_msg:
+        msg = base_msg + msg
+    _chat_post_message(chan_post, msg)
+
+
+def _purge_user_config(event):
+    user_id = event.get("user")
+
+    if utils.redis_client.exists(user_id):
+        utils.redis_client.delete(user_id)
+
+    for chan_id in utils.redis_client.scan_iter(match="C*"):
+        chan_conf = json.loads(utils.redis_client.get(chan_id))
+        phrases_users = chan_conf.get("phrases", {})
+
+        stale_phrases = []
+        for phrase, users in phrases_users.items():
+            if user_id in users:
+                users.remove(user_id)
+                if not users:
+                    stale_phrases.append(phrase)
+        for phrase in stale_phrases:
+            del phrases_users[phrase]
+
+        utils.redis_client.set(chan_id, json.dumps(chan_conf))
+
+    _chat_post_message(channel=event.get("channel"), message="I have purged your configuration!")
+
+
+def _update_user_config(event, phrases, channels):
+    user = event.get("user")
+    if not user:
+        logger.error("User is not present in the message description, cannot set the config")
+        return
+
+    user_conf = {"phrases": phrases, "channels": channels}
+    utils.redis_client.set(user, json.dumps(user_conf))
+
+    # stip phrases of quotes representation symbols
+    chan_phrases = [phrase.strip(utils.QUOTE_PREFIX) for phrase in phrases]
+
+    for chan in channels:
+        chan_id = chan.split("|")[0].strip("<").strip("#")
+
+        if not utils.redis_client.exists(chan_id):
+            chan_conf = {
+                "long_id": chan,
+                "phrases": {phrase: [user] for phrase in chan_phrases},
+            }
+        else:
+            chan_conf = json.loads(utils.redis_client.get(chan_id))
+            for phrase in chan_phrases:
+                phrase_users = chan_conf["phrases"].setdefault(phrase, [])
+                if user not in phrase_users:
+                    phrase_users.append(user)
+        utils.redis_client.set(chan_id, json.dumps(chan_conf))
```

Comment:
@mmatviyiv since relation user - phrases - channels is many to many, maybe it would be more efficient to use relational storage here, but i have reused redis, as it had been added earlier as the message queue.